subt/artf_node.py

```diff
@@ -133,13 +162,65 @@ def detect(self, image):
         result = self.detector(img)
         result_cv = self.cv_detector(img)
         if result or result_cv:
-            self.stdout(result, result_cv)  # publish the results independent to detection validity
+            # publish the results independent to detection validity
+            self.stdout(result, result_cv)  # for debugging in case of crash
+            self.publish('debug_result', str(result))
+            self.publish('debug_cv_result', str(result_cv))
             checked_result = check_results(result, result_cv)
             if checked_result:
-                report = result2report(checked_result, self.depth)
+                report = result2report(checked_result, self.depth, self.fx)
                 if report is not None:
                     self.publish('artf', report)
                     self.publish('debug_artf', image)  # JPEG
                     self.publish('debug_depth', self.depth)
 
+
+if __name__ == "__main__":
+    # run specific test based on recorded stdout
+    import argparse
+    from datetime import timedelta
+    from osgar.lib.serialize import deserialize
+    from osgar.logger import LogReader, lookup_stream_id, lookup_stream_names
+    from ast import literal_eval
+
+    parser = argparse.ArgumentParser(description='Test 3D reports')
+    parser.add_argument('logfile', help='OSGAR logfile')
+    parser.add_argument('--time-limit-sec', '-t', help='cut time in seconds', type=float)
+    args = parser.parse_args()
+
+    names = lookup_stream_names(args.logfile)
+    stdout_stream_id = lookup_stream_id(args.logfile, 'detector.stdout')
+    streams = [stdout_stream_id]
+    if 'detector.debug_depth' in names:
+        depth_stream_id = lookup_stream_id(args.logfile, 'detector.debug_depth')
+        streams.append(depth_stream_id)
+    else:
+        depth_stream_id = None
+
+    # dummy depth for loder logfiles
+    row = [5000] * 640
+    depth = np.array([row] * 360, dtype=np.uint16)
+    fx = 554.25469
+    with LogReader(args.logfile,
+                   only_stream_id=streams) as logreader:
+        for time, stream, msg_data in logreader:
+            if args.time_limit_sec is not None:
+                if time.total_seconds() > args.time_limit_sec:
```

Comment:
The two conditions above could be merged into one.

Reply:
OK, simplified